import-scripts/annotation_suite/merge_mafs.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import argparse
+# Script for mergining MAFs
+# Creates one aggregate header taken from column names acorss MAFs
+# Writes out rows while mapping to header
+# NAs and equivalents replaced with ''
+# - sequenced samples header???
+TUMOR_SAMPLE_BARCODE_COLUMN = "Tumor_Sample_Barcode"
+HUGO_SYMBOL_COLUMN = "Hugo_Symbol"
+ENTREZ_GENE_ID_COLUMN = "Entrez_Gene_Id"
+
+def process_datum(val):
+    """ Cleans up datum. """
+    try:
+        vfixed = val.strip()
+    except AttributeError:
+        vfixed = ""
+    if vfixed in ["", "NA", "N/A", None]:
+        return ""
+    return vfixed
+
+def get_file_header(data_file):
+    """
+        Returns header from MAF
+        Assumes file is tab-delimited
+        Assumes header is first non-commented line
+    """
+    header = []
+    with open(data_file, "r") as header_source:
+        for line in header_source:
+            if not line.startswith("#"):
+                header = list(map(str.strip, line.rstrip().split("\t")))
+                break
+    return header
+
+def merge_headers(data_filenames):
+    """
+        Generates a merged header from all input files.
+        * Also ensures that "Hugo_Symbol" and "Entrez_Gene_Id"
+          are the first 2 columns in the merged MAF header.
+    """
+    merged_header = [HUGO_SYMBOL_COLUMN, ENTREZ_GENE_ID_COLUMN]
+    for fname in data_filenames:
+        columns_to_add = [column for column in get_file_header(fname) if column not in merged_header]
+        merged_header.extend(columns_to_add)
+    return merged_header
+
+def merge_input_mafs(input_mafs, output_maf_filename):
+    """ Generates a merged MAF given a list of input MAFs. """
+    merged_header = merge_headers(input_mafs)
+    sequenced_samples = set()
+
+    rows_to_write = []
+    for input_maf in input_mafs:
+        print('Loading data from ' + input_maf)
+        with open(input_maf, "r") as maf:
+            header_processed = False
+            file_header = get_file_header(input_maf)
+            for line in maf:
+                # add comment processing later (sequenced samples header?)
+                if line.startswith("#"):
+                    continue
+                if not header_processed:
+                    header_processed = True
+                    continue
+                # map row values to current header columns
+                mapped_row = dict(zip(file_header, list(map(lambda x: process_datum(x), line.split("\t")))))
+                sample_id = mapped_row[TUMOR_SAMPLE_BARCODE_COLUMN]
+
+                if sample_id:
+                    sequenced_samples.add(sample_id)
```

Comment:
ok, so the merge script at least will add a sequenced_samples header based on what it has seen in the data table. But this does not include sequenced_samples from an input maf header which do not have any variation records in the data table. Again, we should discuss the needs for this header as a team.
Maybe we want to make the output of the sequenced_samples header optional?

Reply:
for this purpose, sequenced sample headers are not supported. The main merge.py script can be used to merge MAF with the sequence sample header.